### DISCUSSION TITLE
xdrip web headers

### DISCUSSION THREAD

**Discussion by gazhay** [NightscoutFoundation/xDrip#2445] [General]
I have enabled the web interface, however with the exception of some apps which ignore the error, trying to fetch/curl GET the url e.g. http://device.17580/pebble, The headers are not returned with '\r' and \'n'. They are only sent with '\n'.

This causes pretty much every node fetch or similar library to throw errors unless you launch node with '--insecure-http-parser'

Would it be possible to add '\r\m' instead of '\n' to each header line in the responses?

**Comment by Navid200** [comment]
@tzachi-dar

**Comment by Navid200** [comment]
I go through the requests and decide who can best respond.  
I don't understand your request.
Can you tell me how you use xDrip, and how anything you do will become easier if your request is accomplished?
Can you tell me what you do now that you will not need to do any longer if your request is approved?

**Reply by gazhay** [reply]
I use xDrip with a libre2, however I wanted to display the latest sugar on screen during zwift cycling.

To do this I used node to fetch the url 
```` http://<phoneip>:<xdripPort>/pebble ````

The return from the GET has invalid headers.
Protocol states headers are followed by \r and \n they seem to only contain \n

Node throws an insecure header violation.
Other programmatic GET libraries do the same. You can force some to ignore the insecure header response, but it seems inadvisable to disable security for a whole node application to deal with this one invalid response.

Would seem the correct solution is to have xdrip response use the CR-LF specified in the protocol?

**Reply by Navid200** [reply]
@gazhay Thanks for reporting this.
Please open an issue: https://github.com/NightscoutFoundation/xDrip/issues/new/choose

Please describe the issue in the first post.  You can copy your second post from here and paste it if you want as the first post in the issue.

Please include a link to this thread in the first post.

**Comment by jamorham** [comment]
@Navid200 yes you can convert this to issue and I will adopt it but its awkward as it might cause compatibility issues if we don't give it as an option but having said that it sounds like it is a bug.
